modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/snapshot/storage/LastSnapshotStorage.scala

```diff
@@ -56,6 +56,11 @@ object LastSnapshotStorage {
 
       def getCombined: F[Option[(Hashed[S], SI)]] = snapshotR.get
 
+      def getCombinedStream: Stream[F, Option[(Hashed[S], SI)]] =
+        Stream
+          .eval[F, Option[(Hashed[S], SI)]](snapshotR.get)
+          .merge(snapshotR.discrete)
```

Comment:
Can you explain this construct?

Reply:
@TheMMaciek @marcinwadon according to: https://www.javadoc.io/doc/co.fs2/fs2-docs_3/3.4.0/fs2/concurrent/Signal.html#discrete:Stream[F,A]

We probably could just
```scala
snapshotR.continuous merge snapshotR.discrete
```

<img width="633" alt="image" src="https://github.com/Constellation-Labs/tessellation/assets/9077743/35513344-f0f7-49d9-b957-8be5c9ea90ac">